### PR TITLE
refactor: Database connect()

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Reduce the complexity of the method.

- refactor
- remove `&`
- fix PHPDoc

```console
$ phploc system/Database/Config.php
```
Before:
```
Cyclomatic Complexity
  Average Complexity per LLOC                     0.52
  Average Complexity per Class                   13.00
    Minimum Class Complexity                     13.00
    Maximum Class Complexity                     13.00
  Average Complexity per Method                   3.00
    Minimum Method Complexity                     1.00
    Maximum Method Complexity                    11.00
```
After:
```
Cyclomatic Complexity
  Average Complexity per LLOC                     0.38
  Average Complexity per Class                   10.00
    Minimum Class Complexity                     10.00
    Maximum Class Complexity                     10.00
  Average Complexity per Method                   2.50
    Minimum Method Complexity                     1.00
    Maximum Method Complexity                     8.00
```

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/Config.php [hunk 1]
```diff
@@ -38,8 +39,9 @@ class Config extends BaseConfig
     /**
      * Creates the default
      *
-     * @param array|string $group     The name of the connection group to use, or an array of configuration settings.
-     * @param bool         $getShared Whether to return a shared instance of the connection.
+     * @param array|BaseConnection|string|null $group     The name of the connection group to use,
```
Comment (MGatner) [inline system/Database/Config.php:42]: Are these newly-supported types? Or a correction of what is actually already acceptable?

File: system/Database/Config.php [hunk 2]
```diff
@@ -53,16 +55,21 @@ public static function connect($group = null, bool $getShared = true)
         if (is_array($group)) {
             $config = $group;
             $group  = 'custom-' . md5(json_encode($config));
-        }
+        } else {
+            /** @var DbConfig $dbConfig */
+            $dbConfig = config('Database');
 
-        $config ??= config('Database');
+            if ($group === null) {
```
Comment (MGatner) [inline system/Database/Config.php:62]: I'm in favor of getting rid `empty()` but we no longer handle `''` which I think is a possibility.

Comment (MGatner) [reply system/Database/Config.php:62]: Actually it looks like this is acceptable: https://3v4l.org/qSBFc

File: system/Database/Config.php (hunk 1 above)
Comment (kenjis) [reply system/Database/Config.php:42]: A correction of what is actually already acceptable.

File: system/Database/Config.php (hunk 2 above)
Comment (kenjis) [reply system/Database/Config.php:62]: The current `develop` code does not support empty string as `$group`.
https://github.com/codeigniter4/CodeIgniter4/blob/020af6d2d2e61dca96797240308bb96bb080a77a/system/Database/Config.php#L60-L62
`$group` will be the default group name.

And `Database::load()` does not accept empty string.
https://github.com/codeigniter4/CodeIgniter4/blob/020af6d2d2e61dca96797240308bb96bb080a77a/system/Database/Config.php#L78
https://github.com/codeigniter4/CodeIgniter4/blob/020af6d2d2e61dca96797240308bb96bb080a77a/system/Database/Database.php#L42-L46

**Review by MGatner — APPROVED** [review]

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**